extension/src/popup/components/WarningMessages/index.tsx

```diff
@@ -365,3 +374,188 @@ export const ScamAssetWarning = ({
     </div>
   );
 };
+
+export const NewAssetWarning = ({
+  domain,
+  code,
+  issuer,
+  image,
+  newAssetFlags,
+  onClose,
+  setErrorAsset,
+}: {
+  domain: string;
+  code: string;
+  issuer: string;
+  image: string;
+  newAssetFlags: NewAssetFlags;
+  onClose: () => void;
+  setErrorAsset: (errorAsset: string) => void;
+}) => {
+  const { t } = useTranslation();
+  const dispatch: AppDispatch = useDispatch();
+  const warningRef = useRef<HTMLDivElement>(null);
+  const { recommendedFee } = useNetworkFees();
+  const networkDetails = useSelector(settingsNetworkDetailsSelector);
+  const publicKey = useSelector(publicKeySelector);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const { isRevocable, isNewAsset, isInvalidDomain } = newAssetFlags;
+
+  // animate entry
+  useEffect(() => {
+    if (warningRef.current) {
+      setTimeout(() => {
+        warningRef.current!.style.bottom = "0";
+      }, 10);
+    }
+  }, [warningRef]);
+
+  const closeOverlay = () => {
+    if (warningRef.current) {
+      warningRef.current.style.bottom = `-${POPUP_HEIGHT}px`;
```

Comment:
so so NIT: we're using `warningRef.current!.` up above, we should probably be consistent here

Reply:
the one's with `!` are actually only needed because of ts linting since the `warning.current.` line of code isn't directly after the `if (warning.current)`. Personally I like it better w/o for unneeded verbosity reasons. Will merge now but lmk if you feel strongly of including them